Source/Provers/SMTLib/SMTLibBatchTheoremProver.cs

```diff
@@ -165,6 +165,10 @@ private async Task<SolverOutcome> CheckSat(CancellationToken cancellationToken)
 
         var outcomeSExp = responseStack.Pop();
         var result = ParseOutcome(outcomeSExp, out var wasUnknown);
+        // Sometimes Z3 responds with "(error ...)" followed by "unknown"
+        if (outcomeSExp.Name is "error" && responseStack.TryPeek(out var sExpr) && sExpr.Name is "unknown") {
+          responseStack.Pop();
+        }
```

Comment:
Would it be better to do this before `ParseOutcome(...)` instead? Or perhaps we should also set `wasUnknown` to true ourselves so we hit `ParseReasonUnknown` below?